python/python/lancedb/query.py

```diff
@@ -202,6 +237,27 @@ class MatchQuery(FullTextQuery):
     def query_type(self) -> FullTextQueryType:
         return FullTextQueryType.MATCH
 
+    def to_dict(self) -> Dict[str, Any]:
```

Comment:
does that mean we are using Lance FtsQuery for actual deserialization, and use LanceDB FullTextQuery for serialization, and we need to make sure they are equivalent? That feels a bit fragile